active/tab/tabcontrol.md

```diff
@@ -0,0 +1,580 @@
+| TabDraggedOutside | Fires when a Tab is dragged outside of the Tab bar. |
+
+| Method | Description |
+|---|---|
+| DependencyObject ContainerFromItem(object item) | Returns the generated container for the specified item. |
+| DependencyObject ContainerFromIndex(Int32 index) | Returns the generated container for the specified index. |
+
+### TabViewItem properties and events
+
+| Property | Description |
+|:-------- |:----------- |
+| Content | The main content that appears in the tab. |
+| Header | The content that appears inside the tab itself.  |
+| HeaderTemplate | Template for the header object. |
+| Icon | Icon for the tab. |
+| IsCloseable | Determines if the tab shows a close button. Default is "True". |
+
+| Event | Description |
+|---|---|
+| TabClosing | Fires when a tab's close button is clicked. |
+
+# API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```
+namespace MU_XC_NAMESPACE
+{
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+enum TabViewItemWidthMode
+{
+    Actual = 0,
+    Equal = 1,
+};
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+runtimeclass TabViewTabClosingEventArgs
+{
+    Object Item { get; };
+
+    Boolean Cancel { get; set; };
+}
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+unsealed runtimeclass TabView : Windows.UI.Xaml.Controls.Control
+{
+    TabView();
+
+    [MUX_DEFAULT_VALUE("winrt::TabViewItemWidthMode::Actual")]
+    [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+    TabViewItemWidthMode TabViewItemWidthMode{ get; set; };
+
+    // TODO: Remove
+    [MUX_DEFAULT_VALUE("true")]
+    Boolean CanCloseTabs{ get; set; };
+
+    [MUX_DEFAULT_VALUE("true")]
+    Boolean CanDragTabs{ get; set; };
+
+    // TODO: Add
+    [MUX_DEFAULT_VALUE("true")]
+    Boolean CanReorderTabs{ get; set; };
+
+    Object TabStripFooterContent{ get; set; };
+    Windows.UI.Xaml.DataTemplate TabStripFooterContentTemplate{ get; set; };
+    
+    Object TabStripHeader{ get; set; };
+    Windows.UI.Xaml.DataTemplate TabStripHeaderTemplate{ get; set; };
+
+    [MUX_DEFAULT_VALUE("true")]
+    Boolean IsAddTabButtonVisible{ get; set; };
+    Windows.UI.Xaml.Input.ICommand AddTabButtonCommand{ get; set; };
+    Object AddTabButtonCommandParameter{ get; set; };
+
+    event Windows.Foundation.TypedEventHandler<TabView, TabViewTabClosingEventArgs> TabClosing;
+
+    // TODO: Update parameter to TabViewAddTabButtonClickEventArgs
+    event Windows.Foundation.TypedEventHandler<TabView, TabViewAddTabButtonClickEventArgs> AddTabButtonClick;
+
+    // TODO: Add
+    event Windows.Foundation.TypedEventHandler<TabView, TabDraggedOutsideEventArgs> TabDraggedOutside;
+
+    // TODO: Update
+    // From ListView
+    [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+    Object TabItemsSource;
+
+    // TODO: Update
+    [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+    Windows.Foundation.Collections.IVector<Object> TabItems{ get; };
+
+    Windows.UI.Xaml.DataTemplate ItemTemplate;
```

Comment:
What does this template apply to? Since a TabViewItem is a ContentControl, does the template apply to the Content? Or to the Header? Or both? (How do you even do both?) If this templates the header, then should it be called the ItemHeaderTemplate?

Reply:
Discussion: Maps TabViewItem.ContentTemplate

Recommend: Rename to 
TabView.TabContentTemplate
~TabView.TabItemContentTemplate~
~TabView.TabViewItemContentTemplate~

Also rename ItemTemplateSelector to TabContentTemplateSelector

Recommend: Don't we need a header template too?

Reply:
@teaP - can you comment on this?

Reply:
When the ItemsSource is set, use the TabItemTemplate to specify properties on the TabViewItem like Header and IconSource. We don’t have separate ContentTemplate and HeaderTemplate, as both can be satisfied by the following syntax:

```
<controls:TabView x:Name="DataBindingTabView" Grid.Row="1">
    <controls:TabView.TabItemTemplate >
        <DataTemplate x:DataType="local:MyData">
            <controls:TabViewItem Header="{x:Bind Header}" IconSource="{x:Bind IconSource}" Content="{x:Bind MyContent}">/>
        </DataTemplate>
    </controls:TabView.TabItemTemplate >
</controls:TabView>
```